agent/src/beerocks/slave/backhaul_manager/wan_monitor.cpp

```diff
@@ -134,7 +134,8 @@ wan_monitor::ELinkState wan_monitor::process()
             struct ifinfomsg *ifi = (struct ifinfomsg *)NLMSG_DATA(hnl);
 
             // Convert the interface index to name
-            if_indextoname(ifi->ifi_index, name);
+            auto iface_index_name = network_utils::linux_get_iface_name(ifi->ifi_index);
+            std::copy_n(iface_index_name.begin(), iface_index_name.length(), name);
```

Comment:
Why do you copy `iface_index_name` to `name`? You need to remove `name` completely.
Remove its declaration from line 133.
Remove the copy of `iface_index_name` to `name` on line 138.
Compare `m_strWanIfaceName` to `iface_index_name`  using `==` operator on line 137.
Print `iface_index_name` instead of `name` on line 146.

Also I would rename `iface_index_name` to `iface_name`.

Reply:
fixed